File: Excel_Sheet_Column_Title.cpp
/*
Given a positive integer, return its corresponding column title as appear in an Excel sheet.

For example:

    1 -> A
    2 -> B
    3 -> C
    ...
    26 -> Z
    27 -> AA
    28 -> AB 
*/

class Solution {
public:
    string convertToTitle(int n) {
        return n == 0 ? "" : convertToTitle(n/26) + static_cast<char>(--n % 26 + 'A');
    }
};

File: Excel_Sheet_Column_Title.swift
/*
Given an integer columnNumber, return its corresponding column title as it appears in an Excel sheet.

For example:

A -> 1
B -> 2
C -> 3
...
Z -> 26
AA -> 27
AB -> 28
...


Example 1:

Input: columnNumber = 1
Output: "A"
Example 2:

Input: columnNumber = 28
Output: "AB"
Example 3:

Input: columnNumber = 701
Output: "ZY"


Constraints:

1 <= columnNumber <= 231 - 1
*/

/*
Solution 1
recursive check number

Time Complexity: O(log_26^columnNumber)
Space Complexity: O(1)
*/
class Solution {
    func convertToTitle(_ columnNumber: Int) -> String {
        if columnNumber == 0 {
            return ""
        } else {
            var num = columnNumber - 1
            let charA = Character("A").asciiValue!
            let index = Int(charA) + num % 26
            return convertToTitle(num/26) + String(Character(UnicodeScalar(index)!))
        }
    }
}
